### PR TITLE
Enumerated storage fields

### PR DESCRIPTION
**Pull request by dmdashenkov**
- Created an `interface` for all the storage field names to be instance of.
 - Enumerated all basic storage fields to be used in the storage implementations.

### CODE REVIEW

**Comment by dmdashenkov** [comment]
@alexander-yevsyukov PTAL

**Comment by dmdashenkov** [comment]
@alexander-yevsyukov PTAL again

File: server/src/main/java/org/spine3/server/event/storage/EventContextField.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+     */
+    context_event_id,
+
+    /**
+     * A field representing the {@link EventContext#getTimestamp() event posting time}.
+     */
+    context_timestamp,
+
+    /**
+     * A field representing the serialized {@link CommandContext}. This is not the only ay to store it, but it's
```
Comment (alexander-yevsyukov) [inline server/src/main/java/org/spine3/server/event/storage/EventContextField.java]: Have `<p>` after summary sentence.

File: server/src/main/java/org/spine3/server/event/storage/EventContextField.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+package org.spine3.server.event.storage;
+
+import org.spine3.SPI;
+import org.spine3.base.CommandContext;
+import org.spine3.base.EventContext;
+import org.spine3.server.event.EventStorage;
+import org.spine3.server.storage.StorageField;
+
+/**
+ * A container for the storage fields specific for the {@link EventStorage} and its implementations and connected to
```
Comment (alexander-yevsyukov) [inline server/src/main/java/org/spine3/server/event/storage/EventContextField.java]: Please wrap the code to 100 lines. We'll standardize on it soon.

Such a long line is harder to read and understand.

File: server/src/main/java/org/spine3/server/event/storage/EventField.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+ * A container for the storage fields specific for the {@link EventStorage} and its implementations.
+ *
+ * @author Dmytro Dashenkov
+ * @see StorageField
+ */
+@SPI
+public enum EventField implements StorageField {
+
+    /**
+     * A field representing the string representation of the  {@link EventStorageRecord#getProducerId() producer ID}
```
Comment (alexander-yevsyukov) [inline server/src/main/java/org/spine3/server/event/storage/EventField.java]: > `... producer ID}`

Use `@linkplain` for such things.

File: server/src/main/java/org/spine3/server/event/storage/EventField.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+     * of an event.
+     *
+     * <p>Typically, it's an ID of the {@link Aggregate} that produced the {@linkplain Event}.
```
Comment (alexander-yevsyukov) [inline server/src/main/java/org/spine3/server/event/storage/EventField.java]: Why `@linkplain` here?

File: server/src/main/java/org/spine3/server/event/storage/EventField.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+     */
+    producer_id,
+
+    /**
+     * A field representing the {@link EventStorageRecord#getEventId() event ID}.
+     *
+     * <p>An event ID is a protobuf-message type, though the {@link EventStorageRecord#getEventId()} returns its string
```
Comment (alexander-yevsyukov) [inline server/src/main/java/org/spine3/server/event/storage/EventField.java]: Please wrap text text to make it easier to read.

File: server/src/main/java/org/spine3/server/event/storage/EventField.java [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    /**
+     * A field representing the string representation of the  {@link EventStorageRecord#getProducerId() producer ID}
+     * of an event.
+     *
+     * <p>Typically, it's an ID of the {@link Aggregate} that produced the {@linkplain Event}.
+     */
+    producer_id,
+
+    /**
+     * A field representing the {@link EventStorageRecord#getEventId() event ID}.
```
Comment (alexander-yevsyukov) [inline server/src/main/java/org/spine3/server/event/storage/EventField.java]: `@linkplain` here would look better.

File: server/src/main/java/org/spine3/server/event/storage/EventField.java [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+     *
+     * <p>An event ID is a protobuf-message type, though the {@link EventStorageRecord#getEventId()} returns its string
+     * representation. This as what is stored in this field indeed.
+     */
+    event_id,
+
+    /**
+     * A field representing the type of an {@linkplain Event} message e.g. as a {@link TypeUrl}. This may be needed for
```
Comment (alexander-yevsyukov) [inline server/src/main/java/org/spine3/server/event/storage/EventField.java]: Please wrap text text to make it easier to read.

File: server/src/main/java/org/spine3/server/storage/EntityField.java [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+
+    /**
+     * A field representing a timestamp in seconds.
+     *
+     * @see Timestamp#getSeconds()
+     */
+    timestamp,
+
+    /**
+     * A field for storing the part of a timestamp representing the amount of nanoseconds which is less then
```
Comment (alexander-yevsyukov) [inline server/src/main/java/org/spine3/server/storage/EntityField.java]: Why cannot we just refer to the `nanos` field of `Timestamp`?

File: server/src/main/java/org/spine3/server/storage/EntityField.java [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+     * {@link org.spine3.protobuf.Timestamps#NANOS_PER_SECOND 10^9}.
+     *
+     * @see Timestamp#getNanos()
+     */
+    timestamp_nanos,
+
+    /**
+     * A field for storing the serialized {@link Message} bytes. This is the way to store objects of custom types within
```
Comment (alexander-yevsyukov) [inline server/src/main/java/org/spine3/server/storage/EntityField.java]: Please wrap text and have paragraph after the summary sentence.

File: server/src/main/java/org/spine3/server/storage/EntityField.java [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+     * a strongly-typed storage (e.g. relational database) and make the read/write operations easy requiring
+     * no reflection.
+     *
+     * @see Message#toByteArray()
+     * @see Any#getValue()
+     */
+    value,
+
+    /**
+     * A field representing a {@link TypeUrl} of a certain {@link Message} type. This field is commonly used in pair
```
Comment (alexander-yevsyukov) [inline server/src/main/java/org/spine3/server/storage/EntityField.java]: Please wrap text and have paragraph after the summary sentence.

File: server/src/main/java/org/spine3/server/storage/EntityField.java (hunk 8 above)
Comment (dmdashenkov) [inline server/src/main/java/org/spine3/server/storage/EntityField.java]: If you're talking about the `timestamp` field, we store __seconds__, not `Timestamp` there. This is more convenient for queries.

**Comment by dmdashenkov** [comment]
@alexander-yevsyukov PTAL again

**Pull request merged**